project/Build.scala

```diff
@@ -1065,39 +1065,66 @@ object Build extends sbt.Build {
 
   // Testing
 
-  val testTagSettings = Seq(
-      testOptions in Test ++= {
+  val testTagSettings = {
+    val testOptionTags = TaskKey[Seq[String]]("testOptionTags",
+        "Task that lists all test options for javaOptions and testOptions.",
+        KeyRanks.Invisible)
+
+    Seq(
+      testOptionTags := {
+        val stage = (scalaJSStage in Test).value
+        def stageTag: Seq[String] = stage match {
+          case FastOptStage => Seq("fastopt-stage")
+          case FullOptStage => Seq("fullopt-stage")
+          case _            => Seq.empty
+        }
+
+        val sems = stage match {
+          case FastOptStage => (scalaJSSemantics in (Test, fastOptJS)).value
+          case FullOptStage => (scalaJSSemantics in (Test, fullOptJS)).value
+        }
+
+        def semTags: Seq[String] = {
```

Comment:
Maybe move back `sems` and `semTags` down below, where they were before. So that the diff is more understandable.